### ISSUE TITLE
support for libcurl's url parsing?

### ISSUE THREAD

**Issue by jleverenz** [alexcrichton/curl-rust#442]
Is there interest in supporting the URL parsing functionality of later libcurls? https://curl.se/libcurl/c/libcurl-url.html

I'd be happy to share some initial work toward support, with a caveat that I'm using it to learn rust along the way.

**Comment by sagebind** [comment]
I'm not sure what the advantage of using that API would be over using something native such as the `url` crate.

This does run into the problem though of backwards compatibility -- we support curl versions older than 7.62.0 so this API, if added, would have to be an opt-in feature.

**Comment by jleverenz** [comment]
Good point, and not knowing how consuming projects are using curl-rust, I don't have a great answer other than a more complete binding for curl's functionality. [This thread](https://github.com/Orange-OpenSource/hurl/issues/233) on the hurl project got me looking into it, but mostly out of curiosity.

Agreed, it'd have to be opt-in.

**Comment by xtexx** [comment]
`url` crate does not support omitting URL schema (:

**Comment by bdbai** [comment]
> the url crate

https://x.com/mitsuhiko/status/1916755071733297234
